clients/src/main/java/org/apache/kafka/clients/admin/AdminClient.java

```diff
@@ -0,0 +1,312 @@
+     * @throws InterruptedException     If the thread receives an InterruptedException before finishing.
+     * @throws TimeoutException         If the thread times out before receiving a result.
+     * @throws KafkaException           If there was an error deleting the topic.
+     */
+    public void deleteTopic(String topicName, EnumSet<DeleteTopicsFlags> flags) throws InterruptedException;
+
+    enum ListTopicsFlags {
+        /**
+         * List internal topics.
+         */
+        INCLUDE_INTERNAL,
+    }
+
+    class TopicInfo {
+        final String name;
+        final boolean internal;
+        final Map<Integer, Try<TopicPartitionInfo>> partitions;
+
+        TopicInfo(String name, boolean internal, Map<Integer, Try<TopicPartitionInfo>> partitions) {
+            this.name = name;
+            this.internal = internal;
+            this.partitions = partitions;
+        }
+
+        public String name() {
+            return name;
+        }
+
+        public boolean internal() {
+            return internal;
+        }
+
+        public Map<Integer, Try<TopicPartitionInfo>> partitions() {
+            return partitions;
+        }
+
+        @Override
+        public String toString() {
+            return "(name=" + name + ", internal=" + internal + ", partitions=" +
+                    Utils.mkString(partitions) + ")";
+        }
+    }
+
+    class TopicPartitionInfo {
+        final int partition;
+        final Node leader;
+        final List<Node> replicas;
+        final List<Node> isr;
+
+        TopicPartitionInfo(int partition, Node leader, List<Node> replicas, List<Node> isr) {
+            this.partition = partition;
+            this.leader = leader;
+            this.replicas = replicas;
+            this.isr = isr;
+        }
+
+        public int partition() {
+            return partition;
+        }
+
+        public Node leader() {
+            return leader;
+        }
+
+        public List<Node> replicas() {
+            return replicas;
+        }
+
+        public List<Node> isr() {
+            return isr;
+        }
+
+        public String toString() {
+            return "(partition=" + partition + ", leader=" + leader + ", replicas=" +
+                Utils.mkString(replicas, ", ") + ", isr=" + Utils.mkString(isr, ", ") + ")";
+        }
+    }
+
+    /**
+     * List topics in the cluster.
+     *
+     * @param flags                     The flags to use on the request.
+     * @return                          A map from topic names to TopicInfo objects or errors.
+     *                                  An error is returned when we are not authorized to describe the given topic.
+     * @throws InterruptedException     If the thread receives an InterruptedException before finishing.
+     * @throws TimeoutException         If the thread times out before receiving cluster metadata.
+     */
+    Map<String, Try<TopicInfo>> listTopics(EnumSet<ListTopicsFlags> flags) throws InterruptedException;
```

Comment:
i still really think that we should have a `listTopics(Set<String> topics, EnumSet<ListTopicsFlags> flags)`, e.g. one thing i've had to do is write code that is basically waits until i have seen our topics replicated/etc. it would be nice to not have to get all if i dont have to, and also nice to batch for me :)

sample code

```
Set<String> topicsIWant = {..important topics};
checkErrors(createTopics(topicsIWant))
while(topicsIWant.size() > 0) {
  for (Try<TopicInfo> ti : listTopics(topicsIWant)) {
    if (!ti.isFailure) topicsIWant.remove(ti.get().name());
  }
}
```